lib/Resolver.js

```diff
@@ -78,6 +78,19 @@ Resolver.prototype.resolve = function resolve(context, path, request, callback)
 	}, null));
 };
 
+Resolver.prototype.doResolveWithErrorLogging = function doResolveWithErrorLogging(type, request, message, callback, options) {
+	var optionsWithoutLog = {
+		stack: options.stack,
+		missing: options.missing
+	};
+
+	this.doResolve(type, request, message, createInnerCallback((err, result) => {
```

Comment:
Tests fail without it that it fails to throw errors with correct messages otherwise. I was on the same train of thought as you!